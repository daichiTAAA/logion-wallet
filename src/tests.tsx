import { ReactElement } from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import reactRenderer, { ReactTestRenderer, ReactTestRendererJSON, act as reactAct } from 'react-test-renderer';

export function render(element: ReactElement): null | ReactTestRendererJSON | ReactTestRendererJSON[] {
    const component = reactRenderer.create(element);
    return component.toJSON();
}

export function renderer(element: ReactElement): ReactTestRenderer {
    return reactRenderer.create(element);
}

export function shallowRender<E extends ReactElement>(element: ReactElement): E {
    const renderer = createRenderer();
    renderer.render(element);
    return renderer.getRenderOutput();
}

export function mockAccount(address: string, name: string) {
    return {
        address,
        meta: {
            name
        }
    };
}

export { reactAct as act };
