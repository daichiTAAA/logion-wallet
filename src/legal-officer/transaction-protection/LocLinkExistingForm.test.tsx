import { shallowRender } from "../../tests";
import LocLinkExistingForm, { FormValues } from "./LocLinkExistingForm";
import { Mock } from "moq.ts";
import { Control, FieldErrors } from "react-hook-form";

describe("LocLinkExistingForm", () => {

    it("renders", () => {
        const control = new Mock<Control<FormValues>>()
        const errors = new Mock<FieldErrors<FormValues>>()
        const tree = shallowRender(<LocLinkExistingForm
            control={ control.object() }
            errors={ errors.object() }
        />)
        expect(tree).toMatchSnapshot()
    })
})
