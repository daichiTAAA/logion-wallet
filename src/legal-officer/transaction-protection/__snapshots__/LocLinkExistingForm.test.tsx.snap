// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocLinkExistingForm renders 1`] = `
<React.Fragment>
  <h3>
    Link this LOC to an existing LOC
  </h3>
  <p>
    Important: after publication, these data will be publicly available on the blockchain.
  </p>
  <FormGroup
    colors={
      Object {
        "background": "#0c163d",
        "borderColor": "#e6007a",
        "foreground": "#ffffff",
      }
    }
    control={
      <Unknown
        control={[Function]}
        name="locId"
        render={[Function]}
        rules={
          Object {
            "maxLength": Object {
              "message": "The LOC ID has at most 39 digits",
              "value": 39,
            },
            "required": "The LOC ID is required",
          }
        }
      />
    }
    id="locLinkExistingLoc"
    label="Please set the LOC ID you would like to link with this LOC:"
  />
</React.Fragment>
`;
